File: Leetcode/383. Ransom Note.ts
function canConstruct(ransomNote: string, magazine: string): boolean {
    if(ransomNote.length > magazine.length) return false;
    let myMap = new Map();
    
    for(let i=0; i<magazine.length; i++) myMap.set(magazine[i], (myMap.get(magazine[i]) ?? 0) + 1);
    
    for(let i=0; i<ransomNote.length; i++) {
        if(!myMap.get(ransomNote[i])) return false;
        myMap.set(ransomNote[i], myMap.get(ransomNote[i])-1);
    }
    
    return true;
};

File: Leetcode/C++/383. Ransom Note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char,int> m;
        for(int i=0; i<magazine.size(); i++) m[magazine[i]]++;
        
        for(int i=0; i<ransomNote.size(); i++) {
            if(m.find(ransomNote[i]) != m.end()) {
                m[ransomNote[i]]--;
                if(!m[ransomNote[i]]) m.erase(ransomNote[i]);
            } else return false;
        }
        
        return true;
    }
};
